src/plugins/query_enhancements/public/plugin.tsx

```diff
@@ -42,133 +42,126 @@ export class QueryEnhancementsPlugin
     { data, usageCollection }: QueryEnhancementsPluginSetupDependencies
   ): QueryEnhancementsPluginSetup {
     const { queryString } = data.query;
-    const pplSearchInterceptor = new PPLSearchInterceptor({
-      toasts: core.notifications.toasts,
-      http: core.http,
-      uiSettings: core.uiSettings,
-      startServices: core.getStartServices(),
-      usageCollector: data.search.usageCollector,
-    });
-
-    const sqlSearchInterceptor = new SQLSearchInterceptor({
-      toasts: core.notifications.toasts,
-      http: core.http,
-      uiSettings: core.uiSettings,
-      startServices: core.getStartServices(),
-      usageCollector: data.search.usageCollector,
-    });
 
-    const pplControls = [pplLanguageReference()];
-    const sqlControls = [sqlLanguageReference()];
+    // Subscribe to updates to track language changes
+    queryString.getUpdates$().subscribe((query) => {
```

Comment:
Oh no, i missed this part. This is risky. We should not be reregistering the language every time the query changes! Did you sanity test discover with this change?

Reply:
You can avoid this logic entirely since you only need query for the selected language, but looking at the code, you dont really need it right? on line 51&52, you can remove `selectedLanguage` since each instance of `LanguageReference` is created separately for that language.

Reply:
fixed. I removed the subscriber